modular/gater/object_handler.go

```diff
@@ -140,12 +140,24 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 		}
 		log.CtxDebugw(reqCtx.Context(), reqCtx.String())
 	}()
-
 	reqCtx, err = NewRequestContext(r)
 	if err != nil {
-		return
+		// check the object's visibility type whether equal to public read
+		authorized, err = g.baseApp.Consensus().VerifyGetObjectPermission(reqCtx.Context(), "",
+			reqCtx.bucketName, reqCtx.objectName)
+		if err != nil {
+			log.CtxErrorw(reqCtx.Context(), "failed to verify authorize for getting public object", "error", err)
+			err = ErrConsensus
+			return
+		}
+		if !authorized {
+			log.CtxErrorw(reqCtx.Context(), "no permission to operate, object is not public")
```

Comment:
fix `"no permission to operate, object is not public"` as `"no permission to operate due to object is not public"`

Reply:
next pr

Reply:
ditto